vru_use_case_scripts/calc_post_encroachment_time

```diff
@@ -0,0 +1,144 @@
+        ]
+    )
+
+    intersect_point = vehicle_path.intersection(pedestrian_path)
+
+    plt.plot(*vehicle_path.xy)
+    plt.plot(*pedestrian_path.xy)
+
+    BOX_SIZE = 2
+    return shapely.geometry.box(
+        intersect_point.x - BOX_SIZE / 2,
+        intersect_point.y - BOX_SIZE / 2,
+        intersect_point.x + BOX_SIZE / 2,
+        intersect_point.y + BOX_SIZE / 2,
+    )
+
+
+def interpolate_trajectory(odometry_df):
+    times = odometry_df["Message Time (ms)"].tolist()
+    tck_x = splrep(times, odometry_df["Map Position X (m)"].tolist(), k=5)
+    tck_y = splrep(times, odometry_df["Map Position Y (m)"].tolist(), k=5)
+
+    return BSpline(*tck_x, extrapolate=False), BSpline(*tck_y, extrapolate=False)
+
+
+def calc_pedestrian_exit_time(pedestrian_odom_df, encroachment_zone):
+    traj_x, traj_y = interpolate_trajectory(pedestrian_odom_df)
+    start_time = pedestrian_odom_df["Message Time (ms)"].iloc[0]
+    end_time = pedestrian_odom_df["Message Time (ms)"].iloc[-1]
+
+    is_in_zone = False
+    for time in np.arange(start_time, end_time, 100):
+        if is_in_zone and not encroachment_zone.contains(
+            Point(traj_x(time), traj_y(time))
+        ):
+            plt.scatter(traj_x(time), traj_y(time))
+            return time
+
+        if not is_in_zone and encroachment_zone.contains(
+            Point(traj_x(time), traj_y(time))
+        ):
+            is_in_zone = True
+
+
+def calc_vehicle_enter_time(vehicle_odom_df, encroachment_zone):
+    traj_x, traj_y = interpolate_trajectory(vehicle_odom_df)
+    start_time = vehicle_odom_df["Message Time (ms)"].iloc[0]
+    end_time = vehicle_odom_df["Message Time (ms)"].iloc[-1]
+
+    for time in np.arange(start_time, end_time, 10):
+        if encroachment_zone.contains(Point(traj_x(time), traj_y(time))):
+            plt.scatter(traj_x(time), traj_y(time))
+            return time
+
+
+def calc_post_encroachment_time():
+    pass
+
+
+parser = ArgumentParser(
+    prog="calc_post_encroachment_time",
+    description="",
+)
+
+parser.add_argument(
+    "--vehicle-odometry-csv",
+    type=Path,
+    help="CSV file containing vehicle odometry data",
+    required=True,
+)
+
+parser.add_argument(
+    "--pedestrian-odometry-csv",
+    type=Path,
+    help="CSV file containing pedestrian odometry data",
+    required=True,
+)
+
+parser.add_argument(
+    "--show-plot",
+    action="store_true"
+)
+
+cli_args = parser.parse_args()
+
+vehicle_odometry_df = parse_vehicle_odometry_from_csv(cli_args.vehicle_odometry_csv)
+
+pedestrian_odometry_df = parse_pedestrian_odometry_from_csv(
+    cli_args.pedestrian_odometry_csv
+)
+
+encroachment_zone = get_encroachment_zone(vehicle_odometry_df, pedestrian_odometry_df)
+
+vehicle_enter_time = calc_vehicle_enter_time(vehicle_odometry_df, encroachment_zone)
+pedestrian_exit_time = calc_pedestrian_exit_time(
+    pedestrian_odometry_df, encroachment_zone
+)
+
+shapely.plotting.plot_polygon(encroachment_zone, add_points=False)
```

Comment:
Do we want to include the plots this generates as examples? Is there any information required for interpreting these plots

Reply:
No, it's more of a debugging tool. I can remove it if you feel it's unnecessary.